### PR TITLE
Update Sql Auditing swager 

### PR DESCRIPTION
**Pull request by lubalibu**
1. Support auditing to storage with Vnet/Firewall rules API- supporting blob storage authentication with managed identity
2. Add List audit API

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
Looks very good to me! Let us wait for the ARM's feedback about this.

**Comment by lubalibu** [comment]
@bashahee , @salexu - can you review?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/DatabaseExtendedAuditingSettingsList.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "api-version": "2017-03-01-preview"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
+            "id": "/subscriptions/00000000-1111-2222-3333-444444444444/resourceGroups/blobauditingtest-6852/providers/Microsoft.Sql/servers/blobauditingtest-2080/databases/testdb",
+            "name": "default",
+            "type": "Microsoft.Sql/servers/databases/extendedAuditingSettings",
```
Comment (majastrz) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/DatabaseExtendedAuditingSettingsList.json:16]: >Microsoft.Sql/servers/databases/extendedAuditingSettings [](start = 21, length = 56)

The resource type does not match the ID.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ServerExtendedAuditingSettingsList.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "resourceGroupName": "blobauditingtest-4799",
+    "serverName": "blobauditingtest-6440",
+    "api-version": "2017-03-01-preview"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
+            "id": "/subscriptions/00000000-1111-2222-3333-444444444444/resourceGroups/blobauditingtest-6852/providers/Microsoft.Sql/servers/blobauditingtest-2080/extendedAuditingSettings/default",
```
Comment (majastrz) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ServerExtendedAuditingSettingsList.json:13]: >default [](start = 187, length = 7)

Can there be more than one? (`default` in other places implies a ARM singleton resource pattern)

Comment (bashahee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ServerExtendedAuditingSettingsList.json:13]: Cannot be more than one.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/DatabaseExtendedAuditingSettingsList.json (hunk 1 above)
Comment (lubalibu) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/DatabaseExtendedAuditingSettingsList.json:16]: fixed

Comment (majastrz) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/DatabaseExtendedAuditingSettingsList.json:16]: The reosurce type is `Microsoft.Sql/servers/databases/extendedAuditingSettings`, which implies a resource id that looks something like this:
`/subscriptions/{id}/resourceGroups/{rg}/providers/Microsoft.Sql/servers/{serverName}/databases/{db}/extendedAuditingSettings/default`

---
In reply to: [402364988](https://github.com/Azure/azure-rest-api-specs/pull/8897#discussion_r402364988) [](ancestors = 402364988)

Comment (lubalibu) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/DatabaseExtendedAuditingSettingsList.json:16]: please see commit https://github.com/Azure/azure-rest-api-specs/pull/8897/commits/569fa9e5ee245e8264cca11bd698e933c97a7b1a

Comment (majastrz) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/DatabaseExtendedAuditingSettingsList.json:16]: Yes, that is the one I'm looking at. Your `id` has `.../databases/testdb/default`. It should be `.../databases/testdb/extendedAuditingSettings/default`

---
In reply to: [402933615](https://github.com/Azure/azure-rest-api-specs/pull/8897#discussion_r402933615) [](ancestors = 402933615)

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**